src/apps/ocioview/ocioview/message_router.py

```diff
@@ -89,32 +93,41 @@ def set_config_updates_allowed(self, allowed: bool) -> None:
             # Rebroadcast last config record
             message_queue.put_nowait(self._prev_config)
 
-    def cpu_processor_updates_allowed(self) -> bool:
-        return self._cpu_processor_updates_allowed
-
-    def set_cpu_processor_updates_allowed(self, allowed: bool) -> None:
-        self._cpu_processor_updates_allowed = allowed
-        if allowed and self._prev_config is not None:
-            # Rebroadcast last config record
-            message_queue.put_nowait(self._prev_config)
-
     def ctf_updates_allowed(self) -> bool:
         return self._ctf_updates_allowed
 
     def set_ctf_updates_allowed(self, allowed: bool) -> None:
         self._ctf_updates_allowed = allowed
-        if allowed and self._prev_proc is not None:
+        if allowed and self._prev_cpu_proc is not None:
             # Rebroadcast last processor record
-            message_queue.put_nowait(self._prev_proc)
+            message_queue.put_nowait(self._prev_cpu_proc)
+
+    def image_updates_allowed(self) -> bool:
+        return self._image_updates_allowed
+
+    def set_image_updates_allowed(self, allowed: bool) -> None:
+        self._image_updates_allowed = allowed
+        if allowed and self._prev_image_array is not None:
+            # Rebroadcast last image record
+            message_queue.put_nowait(self._prev_image_array)
+
+    def processor_updates_allowed(self) -> bool:
+        return self._processor_updates_allowed
+
+    def set_processor_updates_allowed(self, allowed: bool) -> None:
+        self._processor_updates_allowed = allowed
+        if allowed and self._prev_config is not None:
+            # Rebroadcast last config record
+            message_queue.put_nowait(self._prev_config)
```

Comment:
Shall we use properties here?

```python
    @property
    def image_updates_allowed(self) -> bool:
        return self._image_updates_allowed

    @image_updates_allowed.setter
    def image_updates_allowed(self, allowed: bool) -> None:
        self._image_updates_allowed = allowed
        if allowed and self._prev_image_array is not None:
            # Rebroadcast last image record
            message_queue.put_nowait(self._prev_image_array)
```

Reply:
Good point. I will address this in a future PR